tests/pass-dep/libc/prctl.rs

```diff
@@ -0,0 +1,56 @@
+//@ignore-target: windows # No prctl on Windows
+//@ignore-target: apple # No prctl on macOS
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::thread;
+
+fn main() {
+    let long_name = std::iter::once("test_named_thread_truncation")
+        .chain(std::iter::repeat(" yada").take(100))
+        .collect::<String>();
+
+    fn set_thread_name(name: &CStr) -> i32 {
+        cfg_if::cfg_if! {
+            if #[cfg(any(target_os = "linux", target_os = "illumos", target_os = "solaris", target_os = "freebsd"))] {
+                unsafe { libc::pthread_setname_np(libc::pthread_self(), name.as_ptr().cast()) }
+            } else {
+                compile_error!("set_thread_name not supported for this OS")
+            }
+        }
+    }
+
+    fn get_thread_name(name: &mut [u8]) -> i32 {
+        cfg_if::cfg_if! {
+            if #[cfg(any(
+                target_os = "linux",
+                target_os = "illumos",
+                target_os = "solaris",
+                target_os = "freebsd",
+            ))] {
+                unsafe {
+                    libc::pthread_getname_np(libc::pthread_self(), name.as_mut_ptr().cast(), name.len())
+                }
+            } else {
+                compile_error!("get_thread_name not supported for this OS")
+            }
+        }
+    }
+
+    let result = thread::Builder::new().name(long_name.clone()).spawn(move || {
```

Comment:
This is testing setting thread names via the builder. We already have a test for that, you just need to enable it for Android.

We also should have a test directly invoking `prctl`. But that shouldn't use the Builder at all. It can just set/get the main thread name.